app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java

```diff
@@ -69,7 +68,16 @@ public KairosDbServiceImpl(final KairosDbClient kairosDbClient, final MetricsFac
 
     @Override
     public CompletionStage<MetricsQueryResponse> queryMetricTags(final MetricsQuery query) {
-        throw new UnsupportedOperationException();
+        final Metrics metrics = _metricsFactory.create();
+        final Timer timer = metrics.createTimer("kairosService/queryMetricTags/request");
+        // Filter out rollup metric overrides and forward the query
+        return filterRollupOverrides(query)
+                .thenCompose(_kairosDbClient::queryMetricTags)
+                .whenComplete((result, error) -> {
+                    timer.stop();
+                    metrics.incrementCounter("kairosService/queryMetricTags/success", error == null ? 1 : 0);
```

Comment:
I think we usually (or try?) to use snake case in metric names. 

(looks like in this file we haven't -- not a big deal esp if we have dashboards already)

Reply:
We already have dashboards so not changing at this point.